hwy/contrib/unroller/unroller_test.cc

```diff
@@ -0,0 +1,295 @@
+                                 hn::Vec<TT> const& HWY_RESTRICT x,
+                                 ptrdiff_t const& HWY_RESTRICT places) const {
+    // no stores in a reducer
+    return 0;
+  }
+
+  inline ptrdiff_t ReduceImpl(hn::Vec<TT> const& HWY_RESTRICT x,
+                              T* HWY_RESTRICT to) const {
+    const hn::ScalableTag<T> d;
+    (*to) = hn::ReduceSum(d, x);
+    return 1;
+  }
+
+  inline void ReduceImpl(hn::Vec<TT> const& HWY_RESTRICT x0,
+                         hn::Vec<TT> const& HWY_RESTRICT x1,
+                         hn::Vec<TT> const& HWY_RESTRICT x2,
+                         hn::Vec<TT>& HWY_RESTRICT y) const {
+    y = y + x0 + x1 + x2;
+  }
+};
+
+template <typename T>
+struct minv : UnrollerUnit<minv<T>, T, T> {
+  using TT = hn::ScalableTag<T>;
+  TT d;
+
+  inline hn::Vec<TT> Func(ptrdiff_t const& HWY_RESTRICT idx,
+                          hn::Vec<TT> const& HWY_RESTRICT x,
+                          hn::Vec<TT> const& HWY_RESTRICT y) const {
+    return hn::Min(y, x);
+  }
+
+  inline hn::Vec<TT> YInitImpl() const {
+    return hn::Set(d, std::numeric_limits<T>::max());
+  }
+
+  inline hn::Vec<TT> MaskLoadImpl(ptrdiff_t const& HWY_RESTRICT idx,
+                                  T* HWY_RESTRICT from,
+                                  ptrdiff_t const& HWY_RESTRICT places) const {
+    const hn::ScalableTag<ptrdiff_t> di;
+    using TI = hn::TFromD<decltype(di)>;
+    auto mask = hn::FirstN(d, static_cast<size_t>(places));
+    auto maskneg = hn::Not(hn::FirstN(
+        d, static_cast<size_t>(places +
+                               static_cast<ptrdiff_t>(
+                                   UnrollerUnit<minv<T>, T, T>::UnitLanes()))));
+    if (places < 0) mask = maskneg;
+
+    auto def = YInitImpl();
+    return hn::MaskedLoadOr(def, mask, d, from + idx);
+  }
+
+  inline ptrdiff_t StoreImpl(ptrdiff_t const& HWY_RESTRICT idx,
+                             T* HWY_RESTRICT to,
+                             hn::Vec<TT> const& HWY_RESTRICT x) const {
+    // no stores in a reducer
+    return 0;
+  }
+
+  inline ptrdiff_t MaskStoreImpl(ptrdiff_t const& HWY_RESTRICT idx,
+                                 T* HWY_RESTRICT to,
+                                 hn::Vec<TT> const& HWY_RESTRICT x,
+                                 ptrdiff_t const& HWY_RESTRICT places) const {
+    // no stores in a reducer
+    return 0;
+  }
+
+  inline ptrdiff_t ReduceImpl(hn::Vec<TT> const& HWY_RESTRICT x,
+                              T* HWY_RESTRICT to) const {
+    const hn::ScalableTag<T> d;
+    auto minvect = hn::MinOfLanes(d, x);
+    (*to) = hn::ExtractLane(minvect, 0);
+    return 1;
+  }
+
+  inline void ReduceImpl(hn::Vec<TT> const& HWY_RESTRICT x0,
+                         hn::Vec<TT> const& HWY_RESTRICT x1,
+                         hn::Vec<TT> const& HWY_RESTRICT x2,
+                         hn::Vec<TT>& HWY_RESTRICT y) const {
```

Comment:
I think MSVC might complain about restrict on references. A pointer would be safer for the output arg.